File: zigar-runtime/test/doc-examples/bare-union-price.js
// bare-union-price.js
import { getPrice } from './bare-union-price.zig';

const price = getPrice('USD', 123);
console.log(`USD = ${price.USD}`);
try {
  console.log(`PLN = ${price.PLN}`);
} catch (err) {
  console.error(err);
}
console.log(Object.keys(price));

// console output:
// USD = 123
// TypeError: Accessing property pln when usd is active
// [ 'USD', 'EUR', 'PLN', 'MOP' ]

File: zigar-runtime/test/doc-examples/bare-union-price.zig
// bare-union-price.zig
const Currency = enum { EUR, PLN, MOP, USD };
const Price = union {
    USD: i32,
    EUR: i32,
    PLN: i32,
    MOP: i32,
};

pub fn getPrice(currency: Currency, amount: i32) Price {
    return switch (currency) {
        .USD => .{ .USD = amount },
        .EUR => .{ .EUR = amount },
        .PLN => .{ .PLN = amount },
        .MOP => .{ .MOP = amount },
    };
}
